js/filters.js

```diff
@@ -143,14 +143,22 @@ function TypeaheadFilter(elm) {
   Filter.call(this, elm);
 
   var key = this.$body.data('dataset');
+  var allowText = this.$body.data('allow-text') !== undefined;
   var dataset = typeahead.datasets[key];
-  this.typeaheadFilter = new typeaheadFilter.TypeaheadFilter(this.$body, dataset);
+  this.typeaheadFilter = new typeaheadFilter.TypeaheadFilter(this.$body, dataset, allowText);
```

Comment:
One quick note about this - this value is going to be the result of the evaluation of `this.$body.data('allow-text') !== undefined`, not the actual value of `this.$body.data('allow-text'` itself.  I just wanted to be sure that is the intent here and not to use the actual value of the `data-allow-text` attribute.